<!-- Dataset HK_2 from UrbanNav project. -->
<launch>
    <!-- load dataset-specific params -->
    <rosparam command="load" file="$(find gnss_preprocessor)/config/hongkong.yaml" />

    <!-- path of dataset -->
    <param name="roverMeasureFile" type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_2/COM55_200314_084357.obs" />
    <param name="baseMeasureFile" type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_2/hksc0740.20o" />
    <param name="EmpFile" type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_2/BRDM00DLR_S_20200740000_01D_MN.rnx" />
    <param name="SP3File" type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_2/GFZ0MGXRAP_20200740000_01D_05M_ORB.SP3" />
    <param name="clkFile" type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_2/GFZ0MGXRAP_20200740000_01D_30S_CLK.CLK" />
    <param name="ionexFile" type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_2/casg0740.20i" />
    <param name="atxFile" type="string" value="$(find gnss_preprocessor)/dataset/Shared/igs14.atx" />

    <param name="out_folder"      type="string" value="$(find gnss_preprocessor)/dataset/UrbanNav/HK_2/rtklibResult.pos" />

    <node name="gnss_preprocessor_node" pkg="gnss_preprocessor" type="gnss_preprocessor_node" output="screen" />
    

    <!-- Static fake TF transform -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="static_tf_fake" args="0 0 0 0 0 0 map base_link" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="static_tf_fake_m2w" args="0 0 0 0 0 0 map world" />

    <!-- Rosbag recording -->
    <node pkg="rosbag" type="record" name="rosbag_HK_2" args="record -O $(find gnss_preprocessor)/dataset/Processed/UrbanNav_HK_2_preprocessed.bag -a"/> 

</launch>
